Week7/CurrentNews/app/src/main/java/com/florencenjeri/cocktailsrecipe/App.kt

```diff
@@ -2,26 +2,55 @@ package com.florencenjeri.cocktailsrecipe
 
 import android.app.Application
 import android.content.Context
+import androidx.work.*
 import com.florencenjeri.cocktailsrecipe.database.NewsDao
 import com.florencenjeri.cocktailsrecipe.database.NewsDatabase
-import com.florencenjeri.cocktailsrecipe.networking.RemoteApi
-import com.florencenjeri.cocktailsrecipe.networking.buildApiService
+import com.florencenjeri.cocktailsrecipe.database.NewsRepository
+import com.florencenjeri.cocktailsrecipe.network.RemoteApi
+import com.florencenjeri.cocktailsrecipe.network.buildApiService
+import com.florencenjeri.cocktailsrecipe.worker.RefreshDataWorker
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import java.util.concurrent.TimeUnit
 
 class App : Application() {
+
+
     companion object {
         private lateinit var instance: App
-
         fun getAppContext(): Context = instance.applicationContext
         val apiService by lazy { buildApiService() }
-        val newsRepository by lazy { RemoteApi(apiService) }
-       lateinit var newsDao: NewsDao
+        val remoteApi by lazy { RemoteApi(apiService) }
+        lateinit var newsDao: NewsDao
+        val newsRepository by lazy { NewsRepository() }
```

Comment:
Nicely done, you could've also used the lazy delegate for the DAO, and pass in the Daos and whatnot, to the repository, or where you use it!